pkg/utils/parse_arf_result_test.go

```diff
@@ -403,4 +419,179 @@ Server 3.fedora.pool.ntp.org`
+
+			It("Should parse without error", func() {
+				Expect(err).To(BeNil())
+			})
+
+			It("Should have correct missing values list", func() {
+				Expect(missingVals).To(Equal(expMissingVals))
+			})
+
+			It("Should have correct used values list", func() {
+				Expect(usedVals).To(Equal(expUsedVals))
+			})
+		})
+
+		Context("Contents with wrong formatted url-encoded template content", func() {
+			// an extra space is added in url-encoded data
+			MachineConfig := `apiVersion: machineconfiguration.openshift.io/v1
+			kind: MachineConfig
+			spec:
+			  config:
+				ignition:
+				  version: 3.1.0
+				storage:
+				  files:
+				  - contents:
+					  source: data:,{{ test1%20%3A%3D%20%7B%7B.t he_value_1%7D%7D%0Atest2%20%3A%3D%20%7B%7B.the_value_2%7D%7D%0Atest_not_defined%20%3A%3D%20%7B%7B.the_value_not_defined%7D%7D }}
+					mode: 420
+					overwrite: true
+					path: /etc/chrony.conf`
+
+			_, usedVals, missingVals, err := parseValues(MachineConfig, value_dic)
+
+			It("Should parse with error", func() {
+				Expect(err).NotTo(BeNil())
+			})
+
+			It("Should have correct missing values list", func() {
+				Expect(missingVals).To(BeEmpty())
+			})
+
+			It("Should have correct used values list", func() {
+				Expect(usedVals).To(BeEmpty())
+			})
+		})
+
+		Context("Contents with non-url-encoded template content", func() {
+
+			MachineConfig := `apiVersion: machineconfiguration.openshift.io/v1
+			kind: KubeletConfig
+			spec:
+			  kubeletConfig:
+			  	evictionHard:
+				  imagefs.available: {{.var_kubelet_evictionhard_imagefs_available}}`
+
+			expUsedVals := []string{"var-kubelet-evictionhard-imagefs-available"}
+
+			_, usedVals, missingVals, err := parseValues(MachineConfig, value_dic)
+
+			It("Should parse without error", func() {
+				Expect(err).To(BeNil())
+			})
+
+			It("Should have correct missing values list", func() {
+				Expect(missingVals).To(BeEmpty())
+			})
+
+			It("Should have correct used values list", func() {
+				Expect(usedVals).To(Equal(expUsedVals))
+			})
+		})
+
+		Context("Contents with url-encoded template content and non-urlencoded template content", func() {
+			// an extra space is added in url-encoded data
+			MachineConfig := `apiVersion: machineconfiguration.openshift.io/v1
+			kind: MachineConfig
+			spec:
+			  config:
+				ignition:
+				  version: {{.var_version}}
+				storage:
+				  files: {{.var_file}}
+				  - contents:
+					  source: data:,{{ test1%20%3A%3D%20%7B%7B.the_value_1%7D%7D%0Atest2%20%3A%3D%20%7B%7B.the_value_2%7D%7D%0Atest_not_defined%20%3A%3D%20%7B%7B.the_value_not_defined%7D%7D }}
+					mode: 420
+					overwrite: true
+					path: /etc/chrony.conf`
+
+			expUsedVals := []string{"the-value-1", "the-value-2", "var-version"}
+			expMissingVals := []string{"the-value-not-defined", "var-file"}
+			_, usedVals, missingVals, err := parseValues(MachineConfig, value_dic)
+
+			It("Should parse with error", func() {
+				Expect(err).To(BeNil())
```

Comment:
This seems contradictory. The message says that "it should parse with error", yet the test checks that the error should be `nil`

Reply:
> Thanks for adding tests! This is looking much better!

Thanks for the review, I missed that, let me correct it!